src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java

```diff
@@ -44,20 +44,41 @@ void extract(Table table, BigQueryExtractionData extractionData) {
   }
 
   void cleanup(BigQueryExtractionData extractionData) {
-    String dataBucket = extractionData.getDataBucket();
-    String dataKey = extractionData.getDataKey();
-    String format = extractionData.getFormat();
-    String dataUri = extractionData.getDataUri();
+    deleteBucketAndContents(extractionData.getDataBucket());
+  }
+
+  private void deleteBucketAndContents(String dataBucket) {
+    deleteObjectsInBucket(dataBucket);
+    deleteBucket(dataBucket);
+  }
 
-    BlobId blobId = BlobId.of(dataBucket, dataKey + "." + format);
-    boolean suceeded = storage.delete(blobId);
-    if (!suceeded) {
-      log.warn("Could not delete object {}", dataUri);
+  private void deleteBucket(String dataBucket) {
```

Comment:
Swap this method and the next one around, makes it easier to read if they are in the order they are executed as per method above.